internal/graph/dispatch_throttling_check_resolver_test.go

```diff
@@ -0,0 +1,105 @@
+package graph
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+)
+
+type countResolveCheck struct {
+	count *atomic.Uint32
+}
+
+func newCountResolveCheck() *countResolveCheck {
+	return &countResolveCheck{
+		count: &atomic.Uint32{},
+	}
+}
+
+func (c *countResolveCheck) ResolveCheck(ctx context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error) {
+	c.count.Add(1)
+	return &ResolveCheckResponse{Allowed: true}, nil
+}
+
+func (c *countResolveCheck) Close() {}
+
+func (c *countResolveCheck) NumResolveCheck() uint32 {
+	return c.count.Load()
+}
+
+func TestDispatchThrottlingCheckResolver(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	t.Run("below_shaping_level_should_not_be_shaped", func(t *testing.T) {
```

Comment:
We can drop the use of the pseudo-mock and use our existing `MockCheckResolver`. For example,

```go
t.Run("below_shaping_level_should_not_be_shaped", func(t *testing.T) {
	dispatchThrottlingCheckResolverConfig := DispatchThrottlingCheckResolverConfig{
		TimerTickerFrequency: 1 * time.Hour,
		Level:                200,
		Rate:                 8,
	}
	dut := NewDispatchThrottlingCheckResolver(dispatchThrottlingCheckResolverConfig)
	defer dut.Close()

	mockCtrl := gomock.NewController(t)
	t.Cleanup(mockCtrl.Finish)

	mockCheckResolver := NewMockCheckResolver(mockCtrl)
	dut.SetDelegate(mockCheckResolver)

	mockCheckResolver.EXPECT().
		ResolveCheck(gomock.Any(), gomock.Any()).
		Return(&ResolveCheckResponse{Allowed: true}, nil).
		MaxTimes(1)

	dispatchCounter := &atomic.Uint32{}
	_, err := dut.ResolveCheck(context.Background(), &ResolveCheckRequest{DispatchCounter: dispatchCounter})
	require.NoError(t, err)
	require.Equal(t, uint32(1), dispatchCounter.Load())
})
```